packages/react-client/src/hooks/tsClient.ts

```diff
@@ -0,0 +1,211 @@
+      };
+    },
+    [client],
+  );
+
+  const lastSnapshotRef = useRef<FishjamClientState | null>(null);
+
+  const getSnapshot: () => FishjamClientState = useCallback(() => {
+    if (mutationRef.current || lastSnapshotRef.current === null) {
+      const peers = Object.values(client.getRemotePeers()).map(endpointToPeerState);
+      const components = Object.values(client.getRemoteComponents()).map(endpointToPeerState);
+      const localEndpoint = client.getLocalPeer();
+      const localPeer = localEndpoint ? endpointToPeerState(localEndpoint) : null;
+      const isReconnecting = client.isReconnecting();
+
+      lastSnapshotRef.current = {
+        peers,
+        components,
+        localPeer,
+        isReconnecting,
+      };
+      mutationRef.current = false;
+    }
+
+    return lastSnapshotRef.current;
+  }, [client]);
+
+  const state = useSyncExternalStore(subscribe, getSnapshot);
+
+  function addTrack(
+    track: MediaStreamTrack,
+    simulcastConfig: SimulcastConfig = { enabled: false, activeEncodings: [], disabledEncodings: [] },
+    maxBandwidth: TrackBandwidthLimit = 0, // unlimited bandwidth
+  ): Promise<string> {
+    return client.addTrack(track, undefined, simulcastConfig, maxBandwidth);
+  }
+
+  function connect(config: ConnectConfig) {
+    setPeerStatus("connecting");
+    return client.connect({ ...config, peerMetadata: config?.peerMetadata ?? {} });
+  }
+
+  function disconnect() {
+    return client.disconnect();
+  }
+
+  function removeTrack(trackId: string) {
+    return client.removeTrack(trackId);
+  }
+
+  function replaceTrack(trackId: string, track: MediaStreamTrack) {
+    return client.replaceTrack(trackId, track);
+  }
+
+  function getStatistics(selector?: MediaStreamTrack) {
+    return client.getStatistics(selector);
+  }
+
+  function getBandwidthEstimation() {
+    return client.getBandwidthEstimation();
+  }
+
+  function setTrackBandwidth(trackId: string, maxBandwidth: BandwidthLimit) {
+    return client.setTrackBandwidth(trackId, maxBandwidth);
+  }
+
+  function setEncodingBandwidth(trackId: string, encodingId: string, maxBandwidth: BandwidthLimit) {
+    return client.setEncodingBandwidth(trackId, encodingId, maxBandwidth);
+  }
+
+  function setTargetTrackEncoding(trackId: string, encoding: Encoding) {
+    return client.setTargetTrackEncoding(trackId, encoding);
+  }
+
+  function enableTrackEncoding(trackId: string, encoding: Encoding) {
+    return client.enableTrackEncoding(trackId, encoding);
+  }
+
+  function disableTrackEncoding(trackId: string, encoding: Encoding) {
+    return client.disableTrackEncoding(trackId, encoding);
+  }
+
+  return {
+    ...state,
+    peerStatus,
+    addTrack,
+    connect,
+    disconnect,
+    removeTrack,
+    replaceTrack,
+    getStatistics,
+    getBandwidthEstimation,
+    setTrackBandwidth,
+    setEncodingBandwidth,
+    setTargetTrackEncoding,
+    enableTrackEncoding,
+    disableTrackEncoding,
```

Comment:
**Discussion Idea**: If we decide to always use `useCamera`, `useMicrophone`, and `useScreenShare`, shouldn't we remove those methods?

Reply:
The `useFishjamClient` hook is now exported as public, but as you can see above, we decided to hide it after sorting out how to handle tests. 
In this case we might leave those methods as the hook would be an internal one.

Reply:
https://linear.app/swmansion/issue/FCE-458/rename-client-statuses-joined-connected